go/v1/statement.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * Wrapper APIs for in-toto attestation Statement layer protos.
+ */
+
+package v1
+
+import (
+	"errors"
+	
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func NewStatementPb(subject []*ResourceDescriptor,
+	predicateType string,
+	predicate *structpb.Struct) (*Statement, error) {
+	
+	s := &Statement{
+		Type:          "https://in-toto.io/Statement/v1",
+		Subject:       subject,
+		PredicateType: predicateType,
+		Predicate:     predicate,
+	}
+
+	if !IsValidStatement(s) {
+		return nil, errors.New("malformed Statement")
+	}
+	
+	return s, nil
+}
+
+func IsValidStatement(s *Statement) bool {
+	if s.GetType() != "https://in-toto.io/Statement/v1" {
+		return false
+	}
+	
+	if s.GetSubject() == nil || len(s.GetSubject()) == 0 {
+		return false
+	}
+	
+	// check all resource descriptors in the subject
+	subject := s.GetSubject()
+	for _, rd := range subject {
+		if !IsValidResourceDescriptor(rd) {
+			return false
+		}
```

Comment:
There should be a statement specific check here that `digest` is set, actually.

Reply:
Done.